File: app/src/main/java/ch/bailu/aat/gpx/xml_parser/scanner/Tags.java
package ch.bailu.aat.gpx.xml_parser.scanner;

import java.util.ArrayList;
import java.util.Collections;

import ch.bailu.aat.gpx.GpxAttributesStatic;

public class Tags {
    public final ArrayList<GpxAttributesStatic.Tag> list = new ArrayList<>();

    public void clear() {
        list.clear();
    }

    public boolean notEmpty() {
        return list.size() > 0;
    }


    public void sort() {
        Collections.sort(list);
    }

    public void add(String k, String v) {
        list.add(new GpxAttributesStatic.Tag(k, v));
    }

    public GpxAttributesStatic get() {
        return new GpxAttributesStatic(list.toArray(new GpxAttributesStatic.Tag[]{}));
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/xml/parser/scanner/Tags.kt
package ch.bailu.aat_lib.xml.parser.scanner

import ch.bailu.aat_lib.gpx.attributes.GpxAttributesStatic
import ch.bailu.aat_lib.gpx.attributes.Keys.Companion.toIndex

class Tags {
    val list = ArrayList<GpxAttributesStatic.Tag>()
    fun clear() {
        list.clear()
    }

    fun notEmpty(): Boolean {
        return list.size > 0
    }

    fun sort() {
        list.sort()
    }

    fun add(k: String, v: String) {
        list.add(GpxAttributesStatic.Tag(toIndex(k), v))
    }

    fun get(): GpxAttributesStatic {
        return GpxAttributesStatic(list.toArray(arrayOf()))
    }
}
